MCprep_addon/world_tools.py

```diff
@@ -973,6 +974,126 @@ def world_time_update(self, context):
 	return
 
 
+class MCPREP_OT_render_panorama(bpy.types.Operator):
+	"""Render the Panorama images for a texture Pack"""
+	bl_idname = "mcprep.render_panorama"
+	bl_label = "Render Panorama"
+	bl_description = "Render Panorama for texture Pack"
+	bl_options = {'REGISTER', 'UNDO'}
+
+	panorama_resolution = bpy.props.IntProperty(
+		name="Render resolution",
+		description="The resolution of the output images",
+		default=512
+	)
+
+	save_path = bpy.props.StringProperty(
+		name="Output Folder",
+		description="The folder to output the renders to",
+		subtype="DIR_PATH",
+		default="/tmp\\"
+	)
+
+	render_queue = []
+	render_queue_cleanup = []
+
+	camera_data = None
+
+	old_res_x = None
+	old_res_y = None
+	original_cam = None
+
+	def invoke(self, context, event):
+		return context.window_manager.invoke_props_dialog(self, width=400 * util.ui_scale())
+
+	def draw(self, context):
+		self.layout.prop(self, "panorama_resolution")
+		self.layout.prop(self, "save_path")
+
+	def execute(self, context):
+		# Save old Values
+		self.original_cam = bpy.context.scene.camera
+		self.old_res_x = bpy.context.scene.render.resolution_x
+		self.old_res_y = bpy.context.scene.render.resolution_y
+
+		self.camera_data = bpy.data.cameras.new(name="panorama_cam")
+		self.camera_data.angle = math.pi / 2
+
+		pi_half = math.pi / 2
+		self.render_queue.append({
```

Comment:
Without reading too deeply or doing further analysis, I think your issue with crashing may be that these variables and references to self only exist within the context of this instance of the operator run. You probably want to define a global class that gets instances once and referenced through a global variable which holds any queue objects, so they can safely live outside the operator.

If you do it that way, some of these utility functions you define below could be pushed into this helper, non-operator class.

Reply:
Ok, what I gather from everywhere is that render() tries to get the active window with bpy.context.window (well, not really cuz it's C and not py but yk), and that window is somehow set to None at some point during/after the render. Which doesn't happen with EXEC_DEFAULT

So it would appear that I have to pass an [overriding context](https://docs.blender.org/api/current/bpy.ops.html#overriding-context) to the render function. 

But for some ungodly reason, ImportHelper seems to yeet the object reference at some point. 

```
ReferenceError: StructRNA of type MCPREP_OT_render_panorama has been removed
```

So now, instead of having the issue that the window doesn't exist anymore, I have the issue that my whole class doesn't exist anymore, which probably means that I'll have to refactor it into a global variable after all

Reply:
Not sure if this helps, but when I encountered something related myself, I used an approach where for the _first_ render, I used invoke default, but then all following renders used exec default. You can see my [example here](https://github.com/TheDuckCow/blender-community-render/blob/main/community_render.py#L1336).

Reply:
I tried this, after rendering 4 cameras, it crashes my blender. I have a very unstable configuration of Nvidia Drivers on Linux, that don't work reliably regardless, so I will try this as soon as I'm back on a stable (AMD) system. Thank you for the hint tho!

Reply:
I was finding the same, so I skipped the approach. Just assed a new commit which uses a different approach of using timers to offset when the actual render starts, as otherwise I think the render complete callbacks essentially act recursively (render finishing calls a new callback before the end of the prior render finish callback).